src/repositories/users.repository.ts

```diff
@@ -0,0 +1,60 @@
+import { UserEntity } from '../schemas/user.entity.js';
+import { readFile, writeFile } from 'fs/promises';
+import { randomUUID } from "crypto";
+
+const USERS_FILE_PATH = 'src/db/users.json';
+
+async function loadUsers(): Promise<UserEntity[]> {
+  try {
+    const data = await readFile(USERS_FILE_PATH, 'utf-8');
+    return JSON.parse(data);
+  } catch (err) {
+    console.error('Error loading products:', err);
+    return [];
+  }
+};
+
+async function saveUsers(users: UserEntity[]): Promise<void> {
+  try {
+    await writeFile(USERS_FILE_PATH, JSON.stringify(users), "utf-8")
+    console.log(`User saved successfully!`)
+  } catch (error) {
+    console.error('Error saving products:', error);
+  }
+};
+
+export const UserRepository = {
+  getAll: async (): Promise<UserEntity[]> => {
+    const users = await loadUsers()
+    return users;
+  },
+  getOne: async (userId: string): Promise<UserEntity> => {
+    const users = await loadUsers();
+    const user = users.find((u: UserEntity) => u.id === userId);
+    if (!user) {
+      const customError = {
+        status: 404,
+        message: `User with this id: ${ userId } not found`,
+      };
+      console.error('No such user')
+      throw customError;
```

Comment:
For it to be instance of `MyCustomError`, you should create here an new instance of it and throw it.